### PR TITLE
Add survivor

### PR DESCRIPTION
**Pull request by MeDerpWasTaken**
it should hopefully work, haven't tested it yet though. There are also some other minor fixes in here

### CODE REVIEW

File: src/lib/mafia/roles/neutral/Survivor.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import ActionRole from '@mafia/mixins/ActionRole';
+import SurivorFaction from '@mafia/factions/neutral/Survivor';
+import { Defense, NightActionPriority } from '@mafia/managers/NightActionsManager';
+
+export default class Survivor extends ActionRole {
+
+	public name = 'Surivor';
+	public description = 'A neutral character who just wants to live and is afraid to die.';
```
Comment (Soumil07) [inline src/lib/mafia/roles/neutral/Survivor.ts:8]:
```ts
public description = 'You may vest 4 times in a game.';
```

**Review by Soumil07 — CHANGES_REQUESTED** [review]

File: src/lib/mafia/roles/neutral/Survivor.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
+import ActionRole from '@mafia/mixins/ActionRole';
+import SurivorFaction from '@mafia/factions/neutral/Survivor';
+import { Defense, NightActionPriority } from '@mafia/managers/NightActionsManager';
+
+export default class Survivor extends ActionRole {
+
+	public name = 'Surivor';
```
Comment (Soumil07) [inline src/lib/mafia/roles/neutral/Survivor.ts:7]:
```suggestion
	public name = 'Survivor';
```

File: src/lib/mafia/roles/neutral/Survivor.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+		this.vests--;
+	}
+
+	public tearDown() {
+		this.vested = false;
+	}
+
+}
+
+Survivor.aliases.push('Surv');
```
Comment (Soumil07) [inline src/lib/mafia/roles/neutral/Survivor.ts:51]:
```suggestion
Survivor.aliases = ['Surv'];
```

**Review by Soumil07 — CHANGES_REQUESTED** [review]

**Comment by MeDerpWasTaken** [comment]
Added the requested changes

File: src/lib/mafia/roles/neutral/Survivor.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+import { Defense, NightActionPriority } from '@mafia/managers/NightActionsManager';
+
+export default class Survivor extends ActionRole {
+
+	public name = 'Survivor';
+	public description = 'You may vest 4 times in a game.';
+	public faction = new SurivorFaction();
+	public action = 'vest';
+	public actionGerund = 'vesting';
+	public actionText = 'protect yourself at night.';
```
Comment (Soumil07) [inline src/lib/mafia/roles/neutral/Survivor.ts:12]:
```suggestion
	public actionText = 'protect yourself at night';
```

**Review by Soumil07 — CHANGES_REQUESTED** [review]

**Review by Soumil07 — APPROVED** [review]

**Pull request merged**